cmds/task/actors.go

```diff
@@ -13,34 +13,92 @@ import (
 )
 
 // runCancel cancels the runs of a given task.
-func runCancel(credentials *tcclient.Credentials, args []string, out io.Writer, _ *pflag.FlagSet) error {
+func runCancel(credentials *tcclient.Credentials, args []string, out io.Writer, flagSet *pflag.FlagSet) error {
+	noop, _ := flagSet.GetBool("noop")
+	confirm, _ := flagSet.GetBool("confirm")
+
 	q := makeQueue(credentials)
 	taskID := args[0]
 
+
+	if noop {
+		displayNoopMsg("Would cancel", credentials, args)
+		return nil
+	}
+
+	if confirm {
+		var response = confirmMsg("Cancels", credentials, args)
+		if response == "y" {
+			c, err := q.CancelTask(taskID)
+			run := c.Status.Runs[len(c.Status.Runs)-1]
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				return fmt.Errorf("could not cancel the task %s: %v", taskID, err)
+			}
+			fmt.Fprintln(out, getRunStatusString(run.State, run.ReasonResolved))
+			return nil
+		}
+		return nil
+	}
+
 	c, err := q.CancelTask(taskID)
+	run := c.Status.Runs[len(c.Status.Runs)-1]
+
+
 	if err != nil {
```

Comment:
Hm, this code looks exactly like the code above -- that could be modified to only write this code once (by returning early if confirmation is declined)

Reply:
Maybe I'm not getting the idea but confirm is valid only if the user adds flag --confirm. If user doesn't add --confirm the task will cancel without user's confirmation.

Reply:
Right, the behavior is correct, but the same code (`q.CancelTask` and the lines around it) appears twice.  Duplicating code makes later maintenance difficult, because any change to that code must be made twice, and that's easily forgotten.